woocommerce/woocommerce-gutenberg-products-block
# Support cash on delivery payment method in checkout block

~~**Note - this PR branch is based off #2840.** So you'll see all the diffs from #2840 here too. #2840 implements API changes that this PR needs - allow payment methods to show/hide dynamically (based on shipping method in this case) - #2838.~~ Now #2838 is merged, and this has been rebased to `main` 🧶 

Fixes #2623

Adds support for the [core `Cash on Delivery` payment method (gateway)](https://docs.woocommerce.com/document/cash-on-delivery/) in the cart block.

### Screenshots

Currently the tab just appears/disappears. We might want to add a spinner or other visual finesse for this - cc @LevinMedia 

![Screen Capture on 2020-07-10 at 14-32-42](https://user-images.githubusercontent.com/4167300/87110070-44ab3d00-c2ba-11ea-9f66-2b5379d81a6e.gif)

We can handle that as follow up, as part of improving UI for many (> 3) payment gateways.

### How to test the changes in this Pull Request:
1. Go to `WooCommerce > Settings > Payments` and configure `Cash on Delivery` payment gateway.
2. Set up cart & checkout pages with blocks.
3. Add stuff to cart, proceed to checkout.
4. Complete purchase with COD and confirm everything's working as expected.
2. COD has options so merchant can limit it to shippable products and / or specific shipping methods. Experiment with these settings and confirm that COD works correctly, and is only available when appropriate.
5. Confirm that all other payment gateways still work correctly.

Bonus points - set up more complicated shipping (multiple packages, different carriers) & orders and confirm that the correct payment methods are available dependent on selected shipping carrier.

### Changelog

> Support `Cash on Delivery` core payment gateway in the Checkout block. #2831
